pyccel/parser/semantic.py

```diff
@@ -2682,6 +2688,29 @@ def _visit_DottedName(self, expr):
             bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset),
             severity='fatal')
 
+    def _visit_ListExtend(self, expr):
+        list_variable = self._visit(expr.name[0])
+        iterable = expr.name[1].args[0].value
+
+        if isinstance(iterable, PythonList):
+            added_list = self._visit(iterable)
+            store = [ListAppend(list_variable, a) for a in added_list]
+            return CodeBlock(store)
+
+        else:
+            pyccel_stage.set_stage('syntactic')
+            for_target = self.scope.get_new_name('index')
+            arg = FunctionCallArgument(for_target)
+            func_call = FunctionCall('append', [arg])
+            dotted = DottedName(list_variable, func_call)
```

Comment:
This object is supposed to be syntactic but it contains a semantic object (`list_variable`)